### PR TITLE
CLEANUP: Remove xml module usage in bucketGetAcl api

### LINKED ISSUE #269: Remove xml module usage in bucketGetAcl api
Instead of using xml dependency, just use string concatenation for building XML.

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Fix #269
* Add functions in aclUtils.js to generate the XML DOM

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@ironman-machine go

File: lib/utilities/aclUtils.js [hunk 1]
```diff
@@ -233,4 +234,72 @@ aclUtils.sortHeaderGrants =
...
+ */
+aclUtils.createXmlDom = obj => {
+    if (!Array.isArray(obj)) {
+        const key = Object.keys(obj)[0];
+        aclUtils.createEl(`<${key}>`, obj[key], `</${key}>`);
+    } else {
+        obj.forEach(item => {
+            const key = Object.keys(item)[0];
+
+            if (key === 'Grantee') {
```
Comment (alexandre-merle) [inline lib/utilities/aclUtils.js]: why creating a specific case where we can handle attributes for other keys ?

File: lib/utilities/aclUtils.js [hunk 2]
```diff
@@ -233,4 +234,72 @@ aclUtils.sortHeaderGrants =
...
+                const attrArr = item[key][0][Object.keys(item[key][0])];
+                const attr = [];
+
+                Object.keys(attrArr).forEach(item => {
+                    attr.push(` ${item}="${attrArr[item]}"`);
+                });
+
+                aclUtils.createEl(`<${key + attr.join('')}>`, item[key],
+                                  `</${key}>`);
+            } else if (key !== '_attr') {
```
Comment (alexandre-merle) [inline lib/utilities/aclUtils.js]: what about the _attr key ? we do nothing on this case ?

**Comment by bennettbuchanan** [comment]
We have decided that we do not want to build an object with `aclUtils.constructGetACLsJson` to then use it to build the XML DOM. Thus we are going in a different direction, which will be demonstrated in a new pull request referencing this issue.

**Pull request closed without merging**